src/test/java/com/snowflake/kafka/connector/ConnectClusterBaseIT.java

```diff
@@ -0,0 +1,113 @@
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWFLAKE_URL;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWFLAKE_USER;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.KEY_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.sink.SinkConnector.TOPICS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class ConnectClusterBaseIT {
+
+    protected static EmbeddedConnectCluster connectCluster;
+
+    protected static final String TEST_TOPIC = "kafka-int-test";
+    protected static final String TEST_CONNECTOR_NAME = "test-connector";
+    protected static final Integer TASK_NUMBER = 1;
+    private static final Duration CONNECTOR_MAX_STARTUP_TIME = Duration.ofSeconds(20);
+
+    @Before
+    public void setup() {
+        if (connectCluster == null) {
+            connectCluster = new EmbeddedConnectCluster.Builder().name("kafka-push-connector-connect-cluster").numWorkers(3).build();
+            connectCluster.start();
+            connectCluster.kafka().createTopic(TEST_TOPIC);
+            connectCluster.configureConnector(TEST_CONNECTOR_NAME, createProperties());
+            await().timeout(CONNECTOR_MAX_STARTUP_TIME).until(this::isConnectorRunning);
+        }
+        SnowflakeFakeSinkTask.resetRecords();
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        if (connectCluster != null) {
+            connectCluster.stop();
+            connectCluster = null;
+        }
+    }
+
+    @After
+    public void after() {
+        SnowflakeFakeSinkTask.resetRecords();
+    }
+
+    @Test
+    public void connectorShouldConsumeMessagesFromTopic() {
+        connectCluster.kafka().produce(TEST_TOPIC, "test1");
+        connectCluster.kafka().produce(TEST_TOPIC, "test2");
+
+        await().untilAsserted(() -> {
+            List<SinkRecord> records = SnowflakeFakeSinkTask.getRecords();
+            assertThat(records).hasSize(2);
+            assertThat(records.stream().map(SinkRecord::value)).containsExactly("test1", "test2");
+        });
+    }
+
+    protected Map<String, String> createProperties() {
+        Map<String, String> config = new HashMap<>();
+
+        // kafka connect specific
+        config.put(CONNECTOR_CLASS_CONFIG, SnowflakeFakeSinkConnector.class.getName());
+        config.put(TOPICS_CONFIG, TEST_TOPIC);
+        config.put(TASKS_MAX_CONFIG, TASK_NUMBER.toString());
+        config.put(KEY_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+        config.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
```

Comment:
I expect we'll need to test a connector with different event formats. 
Doesn't need to be addressed right now. Just to keep it in mind.